### PR TITLE
Add the Vue UI (#11)

### PR DESCRIPTION
**Pull request by heyodai**
## Description

- Add a front-end chat interface using Vue and Vuetify. 
- Creates a fake API connecting to an LLM for demo purposes.

## Type of change 

- [ ] Bug Fix 
- [x] New Feature 
- [ ] Documentation 
- [ ] Enhancements 

## How has this been tested?

It hasn't.

## Checklist 

- [x] My code follows the style and contributing guidelines of this repo
- [x] My code has been tested and will not break anything that is existing

### CODE REVIEW

**Comment by heyodai** [comment]
@Semir-Hot01 The critical things to look at are:

- [ChatComponent.vue](https://github.com/heyodai/cs-5588-team-gamestop/blob/5686b9236f7491a2b193eef5191f26919bbea4cb/front/src/components/ChatComponent.vue): This is what creates the custom chat UI on the front end.
- [test-api.py](https://github.com/heyodai/cs-5588-team-gamestop/blob/5686b9236f7491a2b193eef5191f26919bbea4cb/front/test-api.py): This is a simple API in Python to connect the front end to a model.

The rest is mostly boilerplate, although I can walk you through it later if you're curious

File: front/test-api.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    Generate a response from the assistant using the Ollama API.
+
+    TODO: Make the assistant's role and purpose configurable.
+    """
+    # Provide a system message to introduce the assistant
+    msg_list = [
+        {
+            "role": "system",
+            "content": "You are an informative and helpful stock investment assistant. Your purpose is to provide data-driven insights and balanced advice to users seeking information on stock investments. You should focus on offering clear, factual, and practical investment tips, taking into account historical market data and current trends. Remember, your advice should not be construed as financial advice from a certified expert but as guidance to help users make informed decisions. Always encourage users to consult with a financial advisor for personalized advice.",
```
Comment (Semir-Hot01) [inline front/test-api.py:39]: This is really cool

File: front/test-api.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    message.id = str(uuid.uuid4())
+    messages.append(message)
+
+    # Generate a response from the assistant
+    response_text = get_assistant_response()
+
+    # Create and append assistant message
+    assistant_message = Message(
+        id=str(uuid.uuid4()), text=response_text, author="assistant"
+    )
```
Comment (Semir-Hot01) [inline front/test-api.py:79]: I think this still has alot of testing to go and fine tuning but looks good so far

**Review by Semir-Hot01 — APPROVED** [review]
Good work, bring it in and we will build off it as semester rolls forward

File: front/test-api.py (hunk 1 above)
Comment (heyodai) [reply front/test-api.py:39]: And I think we're only scratching the surface of what `ollama` can do!

**Pull request merged**